src/mess/drivers/spc1000.c

```diff
@@ -217,10 +224,16 @@ READ8_MEMBER(spc1000_state::iplk_r)
 WRITE8_MEMBER( spc1000_state::cass_w )
 {
 	bool m = BIT(data, 1) ? true : false;
+	bool s = BIT(data, 2) ? true : false;
 	m_cass->output(BIT(data, 0) ? -1.0 : 1.0);
 	if (m && !m_motor)
 		m_cass->change_state(m_cass->get_state() & CASSETTE_MASK_MOTOR ? CASSETTE_MOTOR_ENABLED : CASSETTE_MOTOR_DISABLED, CASSETTE_MASK_MOTOR);
 	m_motor = m;
+	if (s != m_strobe)
```

Comment:
Do you really need to save the old state here? Does m_centronics->write_strobe(!BIT(data, 2)) not work?